euphonic/force_constants.py

```diff
@@ -556,8 +556,7 @@ def _calculate_phonons_at_qpts(
         # expensive phase calculations later
         sc_image_r = get_all_origins(
             np.repeat(n_sc_shells, 3) + 1, min_xyz=-np.repeat(n_sc_shells, 3))
-        sc_origins = np.einsum('ij,jk->ik', sc_image_r,
-                               self.sc_matrix).astype(np.int32)
+        sc_origins = (sc_image_r @ self.sc_matrix).astype(np.int32)

```

Comment:
It should be noted (though this was in the original) that this will truncate rather than round.